wordpress-mobile/WordPress-iOS
# Quick Start for Existing Users: Refactor Quick Start UI data source

Part of #18388

## Description
This PR doesn't introduce any visible changes. It refactors some of the logic behind how we display quick start tasks in both the quick start card and the expanded quick start view.

## Testing Instructions

The following describes a general smoke test over Quick Start UI:
1. Enable quick start for your site
2. Make sure the two sets of tasks ("Customize your site" and "Grow your audience") are displayed correctly
3. Go through a few tasks
4. Make sure progress reflects normally
5. Open up the expanded view modal for both sets of tasks and make sure the tasks and title are displayed correctly
6. Finish tasks under one set
7. Make sure it gets stroked through in the card
8. Open the expanded view, make sure the view shows the "tasks complete" screen correctly.

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

5. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
